internal/postgresql/postgresql.go

```diff
@@ -1040,6 +1068,12 @@ func (p *Manager) Ping() error {
 	return ping(ctx, p.localConnParams)
 }
 
+func (p *Manager) IsInRecovery(timeout time.Duration) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
```

Comment:
maybe I'm missing part of the workflow, it seems like this would be simpler to call WithTimeout once, above line 529 above, and then reuse it for multiple `isInRecovery` calls?

Reply:
I basically copied the shape of other such helper methods. This isn't very perf sensitive since we're looping and sleeping anyway, so I think it's better to maintain style consistency.